Dockerfile

```diff
@@ -151,19 +151,19 @@ VOLUME /tmp
 VOLUME ${ZB_HOME}/data
 VOLUME ${ZB_HOME}/logs
 
-RUN groupadd -g 1000 zeebe && \
-    adduser -u 1000 zeebe --system --ingroup zeebe && \
+RUN groupadd -g 1001 zeebe && \
+    adduser -u 1001 --system --gid 1001 -h ${ZB_HOME} zeebe && \
```

Comment:
> I've also changed the GID to be the same as the UID (no particular need to have them different).

haven't they been the same before? or am I missing something?

Reply:
@megglos we want to  

So this change has 2 goals:
1. Unify the UID/GID across the apps so it's easier to handle on the Helm side.
2. Use a numeric user ID instead of `zeebe`.

This change shouldn't be breaking for the app but on the operational side like K8s (the SRE and Controllers teams have been informed about the change, and I will do it for the Distro team), but in general it is worth adding it to Zeebe release notes.

> haven't they been the same before? or am I missing something?

That was my copy/paste mistake :sweat_smile: (other apps have that, but not Zeebe, so I will remove that line from the issue desc).

Reply:
> This change shouldn't be breaking for the app 

I actually is, as there is an ownership mismatch of data files stored by Zeebe (which is something other apps don't do afaik). In k8s setups this shouldn't be a problem if fsGroup is set accordingly but for non k8s setups, like compose, you will get permission errors when using the new image. Thus we had [this warning](https://docs.camunda.io/docs/self-managed/operational-guides/update-guide/820-to-830/#zeebe-docker-image-now-runs-with-unprivileged-user-by-default) for 8.3 where we switched to run by default with the uid `1000` user.

We would need to add this to the to be created 8.3 to 8.4 migration guide then.

Reply:
Makes sense :+1:

Reply:
So the conclusion is we add a breaking change note specifying that users upgrading need to explicitly set their UID to 1000 to avoid file permission issues?

Reply:
@npepinpe I'd say we need to tell the users to use the new ID.

So either they change the permission on the disk by running a one-time command like:
```
find / -user 1000 -exec chown 1001 {} \;
```

Or if they don't want to do that, and they are using K8s, they can explicitly mount the filesystem with the new ID using `fsGroup` (I will document that part in the Helm upgrade guide).

Reply:
This has to be documented for Zeebe as well for anyone running self-managed but not using the Helm chart though; at the very least for Docker set ups. Would you still recommend they do the same thing?

Reply:
Yes.

Reply:
@aabouzaid will you take the doc part over?

Reply:
@megglos what is the process in Zeebe for minor release docs and migration guide?

Reply:
We need to reflect it in the following sections of the next/8.4 docs:
- the migration guide, as done [here for 8.2 -> 8.3 ](https://docs.camunda.io/docs/next/self-managed/operational-guides/update-guide/820-to-830/#zeebe-docker-image-now-runs-with-unprivileged-user-by-default)
- create an announcement like done [here for the 8.3 release](https://docs.camunda.io/docs/next/reference/announcements/#camunda-83)

Reply:
are you taking this @aabouzaid ?

Reply:
@aabouzaid let's sync on that on slack :+1:

Reply:
@megglos All clear, I will take it.

Reply:
I'm a bit concerned about this tbh. I feel we trade breaking changes here quite easily, but this is a severe thing to do, especially for our customers which running on bare metal, right? No wonder why people are then reluctant to update. I'm not sure whether it was worth, the benefits over the pain of having this.